pom.xml

```diff
@@ -100,7 +100,12 @@
         <dependency>
             <groupId>org.apache.commons</groupId>
             <artifactId>commons-lang3</artifactId>
-            <version>3.12.0</version>
+            <version>3.13.0</version>
+        </dependency>
+        <dependency>
```

Comment:
? why do we bring this in?

Reply:
We get an upperbound error if we leave it as is:

```
Require upper bound dependencies error for org.apache.commons:commons-lang3:3.12.0 paths to dependency are:
+-io.jenkins.plugins:atlassian-bitbucket-server-integration:4.0.1-SNAPSHOT
  +-org.apache.commons:commons-lang3:3.12.0
and
+-io.jenkins.plugins:atlassian-bitbucket-server-integration:4.0.1-SNAPSHOT
  +-org.apache.commons:commons-text:1.11.0
    +-org.apache.commons:commons-lang3:3.13.0
and
+-io.jenkins.plugins:atlassian-bitbucket-server-integration:4.0.1-SNAPSHOT
  +-org.jenkins-ci.plugins:junit:1252.vfc2e5efa_294f
    +-io.jenkins.plugins:checks-api:2.0.2 (managed) <-- io.jenkins.plugins:checks-api:2.0.2
      +-io.jenkins.plugins:commons-lang3-api:3.13.0-62.v7d18e55f51e2 (managed) <-- io.jenkins.plugins:commons-lang3-api:3.13.0-62.v7d18e55f51e2
        +-org.apache.commons:commons-lang3:3.13.0
```

Reply:
We have to declare it here to resolve an upper bound dependency error: 

```
Require upper bound dependencies error for org.apache.commons:commons-text:1.10.0 paths to dependency are:
+-io.jenkins.plugins:atlassian-bitbucket-server-integration:4.0.1-SNAPSHOT
  +-org.apache.commons:commons-lang3:3.13.0
    +-org.apache.commons:commons-text:1.10.0
and
+-io.jenkins.plugins:atlassian-bitbucket-server-integration:4.0.1-SNAPSHOT
  +-org.jenkins-ci.plugins:junit:1252.vfc2e5efa_294f
    +-io.jenkins.plugins:checks-api:2.0.2 (managed) <-- io.jenkins.plugins:checks-api:2.0.2
      +-io.jenkins.plugins:commons-text-api:1.11.0-95.v22a_d30ee5d36 (managed) <-- io.jenkins.plugins:commons-text-api:1.10.0-68.v0d0b_c439292b_
        +-org.apache.commons:commons-text:1.11.0
```

Reply:
Right, so rather than just adding random dependencies on things we actually do not need or use can we just exclude the import like so:
```xml
 <dependency>
            <groupId>org.apache.commons</groupId>
            <artifactId>commons-lang3</artifactId>
            <version>3.13.0</version>
            <exclusions>
                <exclusion>
                    <!-- commons-text comes in via the jenkins:checks-api at a more modern version, exclude this one -->
                    <groupId>org.apache.commons</groupId>
                    <artifactId>commons-text</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

```

That should make the enforcer happy, me happy and the build green